[Mac] Linking intermittently fails to find openssl symbols

It occurs roughly 30% of the time on Travis. It can be repro locally.

```
note: Undefined symbols for architecture x86_64:
  "_EVP_PKEY_derive_init", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_derive_shared_key::h175d3dfd94a1d47a in foxbox.0.o
  "_EVP_PKEY_derive_set_peer", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_derive_shared_key::h175d3dfd94a1d47a in foxbox.0.o
  "_EVP_PKEY_derive", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_derive_shared_key::h175d3dfd94a1d47a in foxbox.0.o
  "_EVP_PKEY_keygen", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_key_pair::h143d2d5e829af360 in foxbox.0.o
  "_EVP_PKEY_CTX_new", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_key_pair::h143d2d5e829af360 in foxbox.0.o
      foxbox::adapters::webpush::crypto::ecdh_derive_shared_key::h175d3dfd94a1d47a in foxbox.0.o
  "_EVP_PKEY_keygen_init", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_key_pair::h143d2d5e829af360 in foxbox.0.o
  "_EVP_PKEY_CTX_free", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_params::hb8c098bbe81f8a8c in foxbox.0.o
      foxbox::adapters::webpush::crypto::ecdh_generate_key_pair::h143d2d5e829af360 in foxbox.0.o
      foxbox::adapters::webpush::crypto::ecdh_derive_shared_key::h175d3dfd94a1d47a in foxbox.0.o
  "_EVP_PKEY_CTX_ctrl", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_params::hb8c098bbe81f8a8c in foxbox.0.o
  "_EVP_PKEY_CTX_new_id", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_params::hb8c098bbe81f8a8c in foxbox.0.o
  "_EVP_PKEY_paramgen", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_params::hb8c098bbe81f8a8c in foxbox.0.o
  "_EVP_PKEY_paramgen_init", referenced from:
      foxbox::adapters::webpush::crypto::ecdh_generate_params::hb8c098bbe81f8a8c in foxbox.0.o
ld: symbol(s) not found for architecture x86_64

clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Example: https://travis-ci.org/fxbox/foxbox/jobs/124982733#L460

Rusting up to the latest nightly wouldn't solve the problem. See [this job](https://travis-ci.org/fxbox/foxbox/jobs/124967110) for instance

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,6 +20,7 @@ matrix:
   fast_finish: true
   allow_failures:
     - rust: nightly
+    - os: osx
   include:
     - sudo: required
       os: linux

